Transaction/DLC_Funding_Transaction.md

```diff
@@ -0,0 +1,109 @@
+# DLC Funding Transaction
+
+This details the exact format of the on-chain funding transaction, which both sides need to agree on to ensure signatures are valid.
+
+# Table of Contents
+
+* [Transaction](#transaction)
+  * [Transaction Input and Output Ordering](#transaction-input-and-output-ordering)
+  * [Funding Inputs](#funding-inputs)
+  * [FundingOutput](#funding-output)
+  * [Change Outputs](#change-outputs)
+* [Fees](#fees)
+
+# Transaction
+
+## Transaction Input and Output Ordering
+
+The Offerer/Initiator's inputs come first and in the order that they were listed in the Offer message, the Accepter's inputs come after this in the order they are listed in the Accept message. The funding output comes first, followed by the Offerer's change output if applicable, and then the Accepter's change output if applicable.
+
+Note that this will likely change in the future.
+
+## Funding Inputs
+
+All funding inputs must be Segwit or nested P2SH(Segwit) in order to protect against malleability attacks.
+
+## Funding Output
+
+* The funding output script is a P2WSH to:
+
+`2 <offer_funding_pubkey> <accept_funding_pubkey> 2 OP_CHECKMULTISIG`
+
+* Where both `pub_key`s are in compressed format.
+
+## Change Outputs
+
+The funding transaction's change outputs should pay to the address specified in the relevant offer/accept message. A change output's value should equal the total funding amount of that party subtracted by their total collateral as well as their fees for both this transaction as well as their fees for the largest possible Contract Execution Transaction. If this value is below the dust limit, then no change output is added for that party.
```

Comment:
> If this value is below the dust limit

Should say what it is, `1000 sats` I believe

Reply:
I did this and left a comment on #11 so that we can improve this in the future

Reply:
I'm ok to leave a constant for now, but 1000 seem a bit high. I think 500 would be good enough (I just worried that considering that you can get 10000 from a testnet faucet 1000 feels a bit a lot). Also just as a note, we've basically copy pastad the code from bitcoin core for determining if an output is dust: https://github.com/bitcoin/bitcoin/blob/6d8543504d8c5bde1d12a3c60407dee44d2c8e11/src/policy/policy.cpp#L14

Reply:
In Bitcoin-S [we have a fixed threshold of 1000 for dust](https://github.com/bitcoin-s/bitcoin-s/blob/master/core/src/main/scala/org/bitcoins/core/policy/Policy.scala#L63), however, I do agree we should move to deterministic.

Reply:
I see, well if we just put 1000 as a placeholder for now I don't really mind, but just I will probably not change our implementation to that ;)